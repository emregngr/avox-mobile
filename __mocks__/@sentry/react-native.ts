import type { ComponentType } from 'react'

const mockedSentry = {
  init: jest.fn(),
  wrap: <T extends ComponentType<any>>(Component: T): T => Component,
  withScope: jest.fn((callback: (scope: any) => void) =>
    callback({
      setTag: jest.fn(),
      setLevel: jest.fn(),
      setUser: jest.fn(),
      setContext: jest.fn(),
    }),
  ),
  captureException: jest.fn(),
  captureMessage: jest.fn(),
  addBreadcrumb: jest.fn(),
  configureScope: jest.fn(),
  setUser: jest.fn(),
  setTag: jest.fn(),
  setContext: jest.fn(),
  setRelease: jest.fn(),
  setDist: jest.fn(),
  setEnvironment: jest.fn(),
  close: jest.fn(() => Promise.resolve(true)),
  flush: jest.fn(() => Promise.resolve(true)),
}

module.exports = mockedSentry

export type mockedSentryType = typeof mockedSentry
