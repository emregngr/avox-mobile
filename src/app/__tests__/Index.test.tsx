import { render } from '@testing-library/react-native'
import React from 'react'

import Index from '@/app/index'

jest.mock('@/components/common', () => {
  const { View } = require('react-native')

  return {
    FullScreenLoading: (props: any) => <View {...props} testID="full-screen-loading" />,
  }
})

describe('Index Screen', () => {
  it('should render the FullScreenLoading component', () => {
    const { getByTestId } = render(<Index />)

    const loadingComponent = getByTestId('full-screen-loading')
    expect(loadingComponent).toBeTruthy()
  })
})

describe('Index Screen Snapshot', () => {
  it('should render the Index Screen successfully', () => {
    const { toJSON } = render(<Index />)

    expect(toJSON()).toMatchSnapshot()
  })
})
