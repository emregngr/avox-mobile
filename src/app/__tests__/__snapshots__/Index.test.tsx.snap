// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Index Screen Snapshot should render the Index Screen successfully 1`] = `
<View
  testID="full-screen-loading"
/>
`;
